File: Java/150.java
import java.util.*;

class Solution {
    public int evalRPN(String[] tokens) {
        Deque<Integer> stack = new ArrayDeque<>();
        int op1, op2;
        for (String i : tokens) {
            switch (i) {
                case "+":
                    stack.push(stack.pop() + stack.pop());
                    break;
                case "-":
                    op2 = stack.pop();
                    op1 = stack.pop();
                    stack.push(op1 - op2);
                    break;
                case "/":
                    op2 = stack.pop();
                    op1 = stack.pop();
                    stack.push(op1 / op2);
                    break;
                case "*":
                    stack.push(stack.pop() * stack.pop());
                    break;
                default:
                    stack.push(Integer.parseInt(i));
            }
        }
        return stack.pop();
    }
}

File: Python/150.py
from typing import List


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            match token:
                case "+":
                    stack.append(stack.pop() + stack.pop())
                case "-":
                    op1 = stack.pop()
                    op2 = stack.pop()
                    stack.append(op2 - op1)
                case "*":
                    stack.append(stack.pop() * stack.pop())
                case "/":
                    op1 = stack.pop()
                    op2 = stack.pop()
                    stack.append(int(float(op2) / op1))
                case _:
                    stack.append(int(token))
        return stack[0]
